aptos-node/src/lib.rs

```diff
@@ -181,6 +182,70 @@ impl AptosNodeArgs {
     }
 }
 
+fn load_remote_config(initial_config: &NodeConfig) -> anyhow::Result<Option<NodeConfig>> {
+    let data_dir = initial_config.get_data_dir();
+    let config_dir = data_dir.join("remote_configs");
+    if !config_dir.is_dir() {
+        warn!("{:?} is not a dir. using initial config", config_dir);
+        return Ok(None);
+    }
+    let mut entries = fs::read_dir(config_dir.clone())?
+        .filter_map(|res| {
+            let Ok(entry) = res else {
+                return None;
+            };
+            let file_name = entry.file_name().into_string().ok()?;
+            if file_name.starts_with("remote_config_v") {
+                return Some(file_name);
+            }
+            None
+        })
+        .collect::<Vec<_>>();
+    entries.sort();
+
+    if entries.is_empty() {
+        warn!("no remote config found in {:?}", config_dir.as_path());
+        return Ok(None);
+    }
+
+    let re = Regex::new(r"^remote_config_v(\d+)\.yaml$")?;
+    let mut max_version = 0;
+    let mut max_config = String::new();
+
+    for entry in &entries {
+        if let Some(cap) = re.captures(entry) {
+            if let Some(match_) = cap.get(1) {
+                if let Ok(version) = match_.as_str().parse::<i32>() {
+                    if version > max_version {
+                        max_version = version;
+                        max_config = entry.clone();
+                    }
+                }
+            }
+        }
+    }
+
+    if max_config.is_empty() {
+        warn!("no latest version of remote config found");
+        return Ok(None);
+    }
+
+    let latest_remote_config_path = config_dir.join(max_config);
+    let remote_config_file = fs::File::open(latest_remote_config_path)?;
+    let remote_config = serde_yaml::from_reader(remote_config_file)?;
```

Comment:
Can we just be super verbose and print out the whole remote config applied here and its version? Might help sanity check for us and operators if configs seem out of sync